src/services/smart-fields-values-injector.js

```diff
@@ -68,16 +77,22 @@ function SmartFieldsValuesInjector(record, modelName, fieldsPerModel, depth = 0)
 
   this.perform = () =>
     P.each(schema.fields, (field) => {
-      if (!record[field.field]) {
+      if (record
+        && record.dataValues
+        && !Object.prototype.hasOwnProperty.call(record.dataValues, field.field)) {
         if (field.get || field.value) {
-          if (isNotRequestedField(modelName, field.field)) {
+          if (isNotRequestedField(requestedField || modelName, field.field)) {
             return null;
           }
 
           return setSmartFieldValue(record, field, modelName);
         }
         if (_.isArray(field.type)) {
-          record[field.field] = [];
+          // NOTICE: when modifying a sequelize record, make sure you pass the same reference
+          //         to record and record.dataValues
+          const emptyArray = [];
+          record[field.field] = emptyArray;
```

Comment:
[THINKING][IGNORABLE] If I'm not mistaken, it will not be possible after this line to call record.hasOrder() (if the field name is `hasOrder`)

Is this acceptable ? What if the user is working on a smart route, use the recordGetter, and is given this record but can't work with this kind of magic accessor ?

Reply:
Absolutely! This will override the magic accessors.
I warned about this the first time I created the solution. That's why I came with another solution using `smartValues` instead: https://github.com/ForestAdmin/forest-express/pull/443 but it add some complexity.
In those days, it was acceptable to override the magic accessor.
Here is your comment about it: https://github.com/ForestAdmin/forest-express/pull/438#pullrequestreview-445762282 ;)

Any thoughts?